### PR TITLE
Improve benchmarking code and Readme

### PR DESCRIPTION
**Pull request by frant-hartm**

### CODE REVIEW

File: trade-monitor/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+```bash
+bin/jet -v submit lib/jet-trade-monitor-4.3-SNAPSHOT.jar localhost:9092 trades latest 5000 1000 100 \
+  10000 AT_LEAST_ONCE results/ 4 
+  4 OBJECT
+```
+
+The parameters are:
+
+```  
+ <bootstrap.servers> <topic> <offset-reset> <maxLagMs> <windowSizeMs> <slideByMs>
```
Comment (cangencer) [inline trade-monitor/README.md]: would it be lot of work to convert these to use something like properties? so there's defaults? picoCli might be ok as well

**Comment by cangencer** [comment]
Also the benchmark should depend on released version I guess, 4.2

**Comment by frant-hartm** [comment]
We use `master` to test against Jet latest snapshot, created 4.2 branch which depends on the 4.2 release.

File: trade-monitor/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+Run the following command from the `trade-monitor` directory:
+ 
+```bash
+mvn package -pl trade-monitor/kafka-trade-producer -am
+```
+
+2. Start the kafka producer
+
+```bash
+cd kafka-trade-producer
```
Comment (emaxerrno) [inline trade-monitor/README.md]: this is actually `big-data-benchmark/trade-monitor/kafka-trade-producer/target`

File: trade-monitor/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+Apache Kafka cluster up and running
+See [Kafka documentation](https://kafka.apache.org/documentation/)
+
+1. Build the benchmark
+
+Run the following command from the `trade-monitor` directory:
+ 
+```bash
+mvn package -pl trade-monitor/kafka-trade-producer -am
```
Comment (emaxerrno) [inline trade-monitor/README.md]: this doesn't cover how to build the jet-trade-monitor

File: trade-monitor/README.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  10000 AT_LEAST_ONCE results/ 4 
+  4 OBJECT
+```
+
+The parameters are:
+
+```  
+ <bootstrap.servers> <topic> <offset-reset> <maxLagMs> <windowSizeMs> <slideByMs>
+ <snapshotIntervalMs> <processingGuarantee> <outputPath> <kafkaParallelism>
+ <sinkParallelism> <messageType>
```
Comment (emaxerrno) [inline trade-monitor/README.md]: i tried it and it didn't work 

```sh
➜  hazelcast-jet-4.2 bin/jet -v submit lib/trade-monitor.jar localhost:9092 trades latest 5000 1000 100 10000 "at-least-once" results 4 1 1 byte
Verbose mode is on, setting logging level to INFO
Submitting JAR 'lib/trade-monitor.jar' with arguments [localhost:9092, trades, latest, 5000, 1000, 100, 10000, at-least-once, results, 4, 1, 1, byte]
Arguments: [localhost:9092, trades, latest, 5000, 1000, 100, 10000, at-least-once, results, 4, 1, 1, byte]
Usage:
  JetTradeMonitor <bootstrap.servers> <topic> <offset-reset> <maxLagMs> <windowSizeMs> <slideByMs> <snapshotIntervalMs> <processingGuarantee> <outputPath> <latencyType> <kafkaParallelism> <sinkParallelism> <messageType>

<processingGuarantee> - "none" or "exactly-once" or "at-least-once"
<messageType> - byte|object

```

Comment (emaxerrno) [reply trade-monitor/README.md]: ah! this worked. 

```
bin/jet -v submit lib/trade-monitor.jar localhost:9092 trades latest 5000 1000 100 10000 AT_LEAST_ONCE results/ 4 4 OBJECT``` 

they have to be CAPS_LOCK'ed not "string" like the error suggests.

File: trade-monitor/README.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+3. Cancel the job
+
+After running the job for required time you can cancel the job by running
+the following command:
+
+```bash
+bin/jet cancel JetTradeMonitor
```
Comment (emaxerrno) [inline trade-monitor/README.md]: i tried this an it seemed to not work also

```
➜  hazelcast-jet-4.2 bin/jet -v cancel JetTradeMonitor
Verbose mode is on, setting logging level to INFO
13:49:25.611 [ INFO] [c.h.i.c.AbstractConfigLocator] Loading configuration '/home/agallego/workspace/hazelcast-jet-4.2/config/hazelcast-client.yaml' from System property 'hazelcast.client.config'
13:49:25.627 [ INFO] [c.h.i.c.AbstractConfigLocator] Using configuration file at /home/agallego/workspace/hazelcast-jet-4.2/config/hazelcast-client.yaml
13:49:25.962 [ INFO] [c.h.c.i.s.ClientInvocationService] hz.client_1 [jet] [4.2] [4.0.1] Running with 2 response threads, dynamic=true
13:49:26.038 [ INFO] [c.h.c.LifecycleService] hz.client_1 [jet] [4.2] [4.0.1] HazelcastClient 4.0.1 (20200409 - e086b9c) is STARTING
13:49:26.039 [ INFO] [c.h.c.LifecycleService] hz.client_1 [jet] [4.2] [4.0.1] HazelcastClient 4.0.1 (20200409 - e086b9c) is STARTED
13:49:26.074 [ INFO] [c.h.c.i.c.ClientConnectionManager] hz.client_1 [jet] [4.2] [4.0.1] Trying to connect to cluster: jet
13:49:26.078 [ INFO] [c.h.c.i.c.ClientConnectionManager] hz.client_1 [jet] [4.2] [4.0.1] Trying to connect to [127.0.0.1]:5701
13:49:26.106 [ INFO] [c.h.c.LifecycleService] hz.client_1 [jet] [4.2] [4.0.1] HazelcastClient 4.0.1 (20200409 - e086b9c) is CLIENT_CONNECTED
13:49:26.107 [ INFO] [c.h.c.i.c.ClientConnectionManager] hz.client_1 [jet] [4.2] [4.0.1] Authenticated with server [192.168.86.112]:5701:76b97f40-89e5-4fa2-bcd8-df4cde98f164, server version: 4.0.1, local address: /127.0.0.1:33409
13:49:26.111 [ INFO] [c.h.i.d.Diagnostics] hz.client_1 [jet] [4.2] [4.0.1] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
13:49:26.113 [ INFO] [c.h.c.i.s.ClientClusterService] hz.client_1 [jet] [4.2] [4.0.1] 

Members [1] {
        Member [192.168.86.112]:5701 - 76b97f40-89e5-4fa2-bcd8-df4cde98f164
}

13:49:26.172 [ INFO] [c.h.c.i.s.ClientStatisticsService] Client statistics is enabled with period 5 seconds.
13:49:26.200 [ INFO] [c.h.c.LifecycleService] hz.client_1 [jet] [4.2] [4.0.1] HazelcastClient 4.0.1 (20200409 - e086b9c) is SHUTTING_DOWN
13:49:26.210 [ INFO] [c.h.c.i.c.ClientConnectionManager] hz.client_1 [jet] [4.2] [4.0.1] Removed connection to endpoint: [192.168.86.112]:5701:76b97f40-89e5-4fa2-bcd8-df4cde98f164, connection: ClientConnection{alive=false, connectionId=1, channel=NioChannel{/127.0.0.1:33409->/127.0.0.1:5701}, remoteEndpoint=[192.168.86.112]:5701, lastReadTime=2020-07-30 13:49:26.206, lastWriteTime=2020-07-30 13:49:26.206, closedTime=2020-07-30 13:49:26.201, connected server version=4.0.1}
13:49:26.211 [ INFO] [c.h.c.LifecycleService] hz.client_1 [jet] [4.2] [4.0.1] HazelcastClient 4.0.1 (20200409 - e086b9c) is CLIENT_DISCONNECTED
13:49:26.214 [ INFO] [c.h.c.LifecycleService] hz.client_1 [jet] [4.2] [4.0.1] HazelcastClient 4.0.1 (20200409 - e086b9c) is SHUTDOWN
com.hazelcast.jet.picocli.CommandLine$ExecutionException: Error while calling command (public void com.hazelcast.jet.server.JetCommandLine.cancel(com.hazelcast.jet.server.JetCommandLine$Verbosity,com.hazelcast.jet.server.JetCommandLine$TargetsMixin,java.lang.String) throws java.io.IOException): com.hazelcast.jet.core.JobNotFoundException: No job with name or id 'JetTradeMonitor' was found
        at com.hazelcast.jet.picocli.CommandLine.execute(CommandLine.java:1207)
        at com.hazelcast.jet.picocli.CommandLine.access$800(CommandLine.java:141)
        at com.hazelcast.jet.picocli.CommandLine$RunAll.handle(CommandLine.java:1416)
        at com.hazelcast.jet.picocli.CommandLine$RunAll.handle(CommandLine.java:1376)
        at com.hazelcast.jet.picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1243)
        at com.hazelcast.jet.picocli.CommandLine.parseWithHandlers(CommandLine.java:1526)
        at com.hazelcast.jet.server.JetCommandLine.runCommandLine(JetCommandLine.java:163)
        at com.hazelcast.jet.server.JetCommandLine.main(JetCommandLine.java:140)
Caused by: com.hazelcast.jet.core.JobNotFoundException: No job with name or id 'JetTradeMonitor' was found
        at com.hazelcast.jet.server.JetCommandLine.getJob(JetCommandLine.java:513)
        at com.hazelcast.jet.server.JetCommandLine.lambda$cancel$1(JetCommandLine.java:255)
        at com.hazelcast.jet.server.JetCommandLine.runWithJet(JetCommandLine.java:457)
        at com.hazelcast.jet.server.JetCommandLine.cancel(JetCommandLine.java:254)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.hazelcast.jet.picocli.CommandLine.execute(CommandLine.java:1189)
        ... 7 more
```

Comment (emaxerrno) [reply trade-monitor/README.md]: I hit control-c and it worked.

File: trade-monitor/README.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+The results can be found in the `results` folder, in files named 0, 1, 2 ...
+up to a number depending on the `sinkParallelism` parameter.
+
+The files contain latencies for individual windows. You can process them further, 
+e.g. to compute percentiles, maximum latency, plot histograms etc..
+
+For example, to simply plot all results you can:
+
+```
+gnuplot -e "plot 'target/0','target/1'; pause -1"
```
Comment (emaxerrno) [inline trade-monitor/README.md]: this also didn't worked. but i think you just have a typo on `s/target/results/g`


```
gnuplot -e "plot 'results/0','results/1'; pause -1"
```

Comment (emaxerrno) [reply trade-monitor/README.md]: I ran this on my laptop, and here are the results but I'm not sure how to interpret. though using [vectorized.io/redpanda](vectorized.io/redpanda)

test jpg

**Pull request merged**